### ISSUE TITLE
Memory region barrier may generate duplicate edges?

### ISSUE THREAD

**Issue by qinsoon** [mmtk/mmtk-core#743]
It seems memory region barrier does not log objects (possibly because it does not know object references).
https://github.com/mmtk/mmtk-core/blob/a272c2399df39177361788449b1f96d060a86080/src/plan/barriers.rs#L215-L222

It is unlike object write barrier.
https://github.com/mmtk/mmtk-core/blob/a272c2399df39177361788449b1f96d060a86080/src/plan/barriers.rs#L203-L213

That means we may push duplicate edges to `ProcessModBuf` through memory region barrier. This will cause our edge logger (enabled by `extreme_assertions`) check fail.

**Comment by qinsoon** [comment]
@wenyuzhao Can you confirm this is a problem (just in case I misunderstood the issue)?

**Comment by wenyuzhao** [comment]
yes we don't log edges, and there will be duplicate edges. But this does not affect correctness.

**Comment by qinsoon** [comment]
Yeah, it does not affect correctness. However, we will need to get the edge logger working. I think we can check root edges and non root edges differently. We can treat edges from `ProcessModBuf` as `roots`, and for plans that use object barrier and `ProcessModBuf`, we allow duplicate root edges.

**Comment by caizixian** [comment]
We discussed that this behaviour is probably reasonable but should be documented
